packages/widgets/src/Pagination/Pagination.js

```diff
@@ -0,0 +1,161 @@
+import React from 'react'
+import propTypes from '@dhis2/prop-types'
+import i18n from '@dhis2/d2-i18n'
+import { Button, SingleSelect, SingleSelectOption } from '@dhis2/ui'
+import { ChevronRight, ChevronLeft } from '@dhis2/ui-icons'
+
+import styles from './Pagination.styles.js'
+
+/**
+ * @module
+ * @param {Pagination.PropTypes} props
+ *
+ * @returns {React.Component}
+ *
+ * @example import { Pagination } from @dhis2/ui
+ * @see Specification: {@link https://github.com/dhis2/design-system/blob/master/molecules/pagination.md|Design system}
+ * @see Live demo: {@link /demo/?path=/story/pagination--default|Storybook}
+ */
+const Pagination = ({
+    dataTest,
+    hidePageSizeSelect,
+    hideGoToPageSelect,
+    page,
+    pageCount,
+    pageSize,
+    total,
+    pageSizes,
+    onPageChange,
+    onPageSizeChange,
+}) => {
+    const firstItem = Math.min((page - 1) * pageSize + 1, total)
+    const lastItem = Math.min(firstItem + pageSize - 1, total)
+    const availablePages = Array.from({ length: pageCount }, (_x, i) =>
+        (i + 1).toString()
```

Comment:
This seems ripe for abstracting into a separate file and then unit testing them.